src/vault.py

```diff
@@ -12,21 +12,104 @@
 import hvac  # type: ignore[import]
 import requests  # type: ignore[import]
 
+CHARM_POLICY_NAME = "local-charm-policy"
+CHARM_ACCESS_ROLE = "local-charm-access"
+
 CHARM_PKI_MP = "charm-pki-local"
 CHARM_PKI_ROLE = "local"
 CHARM_PKI_ROLE_CLIENT = "local-client"
 
+
+CHARM_POLICY = """
+# Allow managment of policies starting with charm- prefix
+path "sys/policy/charm-*" {
+  capabilities = ["create", "read", "update", "delete"]
+}
+
+# Allow discovery of all policies
+path "sys/policy/" {
+  capabilities = ["list"]
+}
+
+# Allow management of approle's with charm- prefix
+path "auth/approle/role/charm-*" {
+  capabilities = ["create", "read", "update", "delete", "list"]
+}
+
+# Allow discovery of approles
+path "auth/approle/role" {
+  capabilities = ["read"]
+}
+path "auth/approle/role/" {
+  capabilities = ["list"]
+}
+
+# Allow charm- prefixes secrets backends to be mounted and managed
+path "sys/mounts/charm-*" {
+  capabilities = ["create", "read", "update", "delete", "sudo"]
+}
+
+# Allow charm- prefixes pki backends to be used
+path "charm-pki-*" {
+  capabilities = ["create", "read", "update", "delete", "list", "sudo"]
+}
+
+# Allow discovery of secrets backends
+path "sys/mounts" {
+  capabilities = ["read"]
+}
+path "sys/mounts/" {
+  capabilities = ["list"]
+}"""
```

Comment:
Consider dropping this into it's own `*.hcl` file?

Reply:
Using the hcl file now. As for the JSON serializing, yeah we may want to do this, not sure if it's worth spending the effort on this now though.